gcc/rust/typecheck/rust-hir-trait-ref.h

```diff
@@ -436,6 +438,9 @@ class AssociatedImplTrait
 
   void setup_associated_types ();
 
+  void setup_associated_types2 (const TyTy::BaseType *self,
```

Comment:
I think the name might be a little unclear. Can we overload the previous `setup_associated_types` or maybe give it a name reflecting the fact that it performs a different operation?

Reply:
This function is designed to overtake the other setup_associated_types the other function is still used in one path for dyn traits which I need to refactor and update for #914 so I would rather fix this then since this PR is already so big.

Reply:
I've raised https://github.com/Rust-GCC/gccrs/issues/1105 to track this